src/components/flashcard-set/flashcard-set.tsx

```diff
@@ -26,70 +29,87 @@ export const FlashcardSet = ({
   onCardChange,
 }: FlashcardSetProps) => {
   const [activeCardKey, setActiveCardKey] = useState(initialActiveCardKey);
+  const [draggingKey, setDraggingKey] = useState('');
+  const dragControls = useDragControls();
+  const dragClass = draggingKey ? 'dragging' : '';
 
   useEffect(() => {
     setActiveCardKey(initialActiveCardKey);
   }, [initialActiveCardKey]);
 
   return (
-    <div className={`c-flashcard-set ${className}`}>
-      <Reorder.Group axis="y" values={cards} onReorder={handleCardReorder}>
+    <div className={`c-flashcard-set ${className} ${dragClass}`}>
+      <Reorder.Group axis="y" values={cards} onReorder={onCardReorder}>
         {getCards()}
       </Reorder.Group>
     </div>
   );
 
   function getCards() {
     return cards.map((card, index) => {
+      const variant = card.key === activeCardKey ? 'active' : 'inactive';
       return (
-        <Reorder.Item key={card.key} value={card}>
+        <Reorder.Item
+          key={card.key}
+          className="card-with-buttons"
+          value={card}
+          onDragStart={(e, info) => onDragStart(e, info, card.key)}
+          onDragEnd={() => setDraggingKey('')}
+          dragControls={dragControls}
+        >
+          <ReorderButtonStrip
+            variant={variant}
+            isDragging={card.key === draggingKey}
+            dragHandle={DRAG_HANDLE}
+            index={index + 1}
+            onUpClick={() => handleUpClick(index)}
+            onDownClick={() => handleDownClick(index)}
+          />
           <Flashcard
             card={card}
-            index={index + 1}
-            variant={getCardVariant(card.key)}
+            variant={variant}
             onFocus={() => setActiveCardKey(card.key)}
-            onCardChange={(card) => variant === 'editable' && onCardChange?.(card)}
-            onDownClick={() => handleDownClick(index)}
-            onUpClick={() => handleUpClick(index)}
-            onRemoveClick={() => variant === 'editable' && onDeleteCardClick?.(card)}
+            onCardChange={onCardChange}
             onSpeakerClick={handleSpeakerClick}
           />
+          <Button onClick={() => onDeleteCardClick(card)} variant="invisible" ariaLabel="trash">
+            <TrashIcon variant={variant} />
+          </Button>
         </Reorder.Item>
       );
     });
   }
 
-  function getCardVariant(cardKey: string) {
-    if (variant === 'readonly') return 'readonly';
-    return activeCardKey === cardKey ? 'active' : 'inactive';
+  function onDragStart(e: any, info: any, key: string) {
+    if (e.target.classList.contains(DRAG_HANDLE)) {
```

Comment:
for my education, if the drag handle lives in the child component's class
how does `e.target.classList` ever contain the handle?

or does `e.target` change depending on where you started dragging -- so this would only set the dragging key if you begin on the icon?